parser/parser.go

```diff
@@ -27,23 +28,32 @@ type QueryField struct {
 
 type QueryFieldList []QueryField
 
-func ParseQueryDir(dir string, schema *ast.Schema) (QueryFieldList, error) {
-	fields := QueryFieldList{}
+func ParseQuerySource(source string, schema *ast.Schema) (QueryFieldList, error) {
+	//check if file or directory
+	fileInfo, err := os.Stat(source)
+	if err != nil {
+		return nil, err
+	}
+
+	if fileInfo.IsDir() {
+		return parseQueryDir(source, schema)
+	} else {
+		return parseQueryList(source, schema)
+	}
+}
+
+func parseQueryDir(dir string, schema *ast.Schema) (QueryFieldList, error) {
 	files := findQueryFiles(dir)
 	// @todo: error out if no files are found
+	return queryTokensFromFiles(files, schema)
+}
 
-	for _, file := range files {
-		doc, err := parseQueryFile(file)
-		if err != nil {
-			return nil, err
-		}
-		// Ignoring errors here as there could be validation errors that we're not interested in.
-		// We only call `.Validate` so the parser can populate the `Definition` fields.
-		_ = gqlvalidator.Validate(schema, doc)
-		fields = buildQueryTokens(doc, fields)
+func parseQueryList(file string, schema *ast.Schema) (QueryFieldList, error) {
+	files, err := getLinesFromFile(file)
```

Comment:
this is a little confusing to me, the function is called `getLines`, however it returns `files`? I'm not following, what are we referring to with `Lines` here?

Reply:
Good point, it is indeed a bit confusing. The queryList is really a list of query files, one per line. I will go around the whole code making a bit more sense of it